subsys/net/l2/openthread/Kconfig

```diff
@@ -32,57 +32,20 @@ menuconfig OPENTHREAD_DEBUG
 	help
 	  This option enables log support for OpenThread
 
-if OPENTHREAD_DEBUG
-
-choice
-	prompt "OpenThread stack log level"
-	help
-	  This option selects log level for OpenThread stack.
-
-config OPENTHREAD_LOG_LEVEL_ERROR
-	bool "Error"
-config OPENTHREAD_LOG_LEVEL_WARNING
-	bool "Warning"
-config OPENTHREAD_LOG_LEVEL_INFO
-	bool "Info"
-config OPENTHREAD_LOG_LEVEL_DEBUG
-	bool "Debug"
-endchoice
-
 endif
 
-endif
-
-config	OPENTHREAD_LOG_LEVEL
-	int
-	default 1 if OPENTHREAD_LOG_LEVEL_ERROR
-	default 2 if OPENTHREAD_LOG_LEVEL_WARNING
-	default 3 if OPENTHREAD_LOG_LEVEL_INFO
-	default 4 if OPENTHREAD_LOG_LEVEL_DEBUG
-	default 0
+module = OPENTHREAD
+module-dep = NET_LOG
+module-str = Log level for OpenThread stack
+module-help = Enables OpenThread stack code to output debug messages.
+source "subsys/net/Kconfig.template.log_config.net"
```

Comment:
Will this work as previously implemented map for level to value mapping?

Reply:
Yes, it is a same thing

Reply:
So my only concern is that OpenThread values may diverge from Zephyr values some day.

Reply:
The problem is that it already diverged (but we do not have this version in Zephyr yet):
https://github.com/openthread/openthread/blob/71d4ad3e777672a441e238344290df302ad86bb7/include/openthread/platform/logging.h#L89

Actually, we should separate OpenThread internal logging configuration from Zephyr logs. I'd say, to keep the OPENTHREAD_LOG_LEVEL as it is, and use it only to configure log level in OpenThread. That shouldn't be difficult, as the only module that uses this config for Zephyr logs which actually prints some logs is `openthread/platform/radio.c` . We could use OPENTHREAD_L2_LOG_LEVEL there instead perhaps?

Reply:
Ok, that makes sense. I will send a new version that restores OPENTHREAD_LOG_LEVEL. I will leave the radio.c changes to you do at later time.